### PR TITLE
Make VTK optional

### PR DESCRIPTION
**Pull request by JDBetteridge**
# Description
Recent issues on ARM based Mac + the long standing issue where we have to distribute VTK wheels prompted making VTK an optional dependency.

### CODE REVIEW

**Comment by colinjcotter** [comment]
Are you still going to distribute the wheels?

**Comment by JDBetteridge** [comment]
Probably, although VTK are getting better at releasing up to date wheels: https://pypi.org/project/vtk/#history

File: firedrake/output/paraview_reordering.py [hunk 1]
```diff
@@ -2,6 +2,14 @@
 import numpy as np
 from pyop2.utils import as_tuple
 
+try:
+    import vtkmodules.vtkCommonDataModel
+    del vtkmodules.vtkCommonDataModel
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Error importing vtkmodules, you may need to install by executing\n\t"
+        "pip install vtk"
```
Comment (connorjward) [inline firedrake/output/paraview_reordering.py:11]: Is this going to be robust enough? What is VTK's release schedule like?

File: firedrake/output/paraview_reordering.py [hunk 2]
```diff
@@ -2,6 +2,14 @@
 import numpy as np
 from pyop2.utils import as_tuple
 
+try:
+    import vtkmodules.vtkCommonDataModel
+    del vtkmodules.vtkCommonDataModel
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Error importing vtkmodules, you may need to install by executing\n\t"
```
Comment (connorjward) [inline firedrake/output/paraview_reordering.py:10]: I feel like we could give the user a slightly better message. Something like:

"Cannot import vtkmodules. Firedrake does not install VTK by default so you may need to install it by running pip install vtk first."

I feel that the current message doesn't sound deliberate enough.

**Review by connorjward — CHANGES_REQUESTED** [review]

File: firedrake/output/paraview_reordering.py (hunk 1 above)
Comment (connorjward) [reply firedrake/output/paraview_reordering.py:11]: I've just checked the link you shared above. I'm convinced. They seem to be releasing wheels every month or so and are supporting the latest Pythons.

**Review by connorjward — COMMENTED** [review]
I am happy with this. @JDBetteridge believes that some bikeshedding/further API thought is required before merging.

**Review by dham — DISMISSED** [review]
Please make vtk default on.

**Comment by JDBetteridge** [comment]
I think this is ready, I think I want to kick the "move `checkpointing.py` into the `output` directory" can down the road.

**Review by connorjward — APPROVED** [review]

**Review by dham — APPROVED** [review]

**Pull request merged**